includes/class-wc-form-handler.php

```diff
@@ -747,12 +747,12 @@ public static function add_to_cart_action( $url = false ) {
 		}
 
 		// If we added the product to the cart we can now optionally do a redirect.
-		if ( $was_added_to_cart && wc_notice_count( 'error' ) === 0 ) {
+		if ( $was_added_to_cart && 0 ===  wc_notice_count( 'error' )  ) {
 			// If has custom URL redirect there
 			if ( $url = apply_filters( 'woocommerce_add_to_cart_redirect', $url ) ) {
 				wp_safe_redirect( $url );
 				exit;
-			} elseif ( get_option( 'woocommerce_cart_redirect_after_add' ) === 'yes' ) {
+			} elseif ( 'yes' ===  get_option( 'woocommerce_cart_redirect_after_add' )  ) {
```

Comment:
@mikejolley

Fix this issue on new pull request https://github.com/woocommerce/woocommerce/pull/16558